db/migrations/03_setting_up_available_nights_table.sql

```diff
@@ -0,0 +1 @@
+CREATE TABLE available_nights (id SERIAL PRIMARY KEY, date DATE, listing_id INTEGER REFERENCES listings (id));
```

Comment:
We think that the listings::available_dates relationship is a many-to-many. This table is set up as a many-to-one (which will cause date duplication in the available_dates table if, for example, you wanted to have multiple properties available on the same date). Perhaps a solution would be to use a join table and so have three in total: listings, dates/calendar, join table (listings_available-dates). We have drawn a little diagram of this in excel if it's more clear and would help!

Reply:
if you could share the diagram please